src/BitBucketOnPrem.ts

```diff
@@ -44,7 +44,11 @@ export const getFileTreeFromBitbucket =
             }
         });
         const fileList: any = await res.json();
-        files = [...files, ...fileList.values];
+        if (Array.isArray(fileList.values)) {
+            files = [...files, ...fileList.values];
+        } else {
+            logger.error("Bitbucket OnPrem files tree request returned an unexpected value", { res, fileList });
```

Comment:
The function should probably return here

Reply:
i'm not sure if i agree
we're in a loop for getting in pieces the files tree from BB.
if a single response returned with broken values (from a reason we do not understand yet), shouldn't we try continue to the next page?

Reply:
And then we will see missing files and customers will contact us but the SEs won't know why only a specific file is missing

Reply:
so return what?
return files will still be missing files,
return empty array is the same